tests/unit/amo/components/TestHeader.js

```diff
@@ -1,74 +1,55 @@
-import I18nProvider from 'core/i18n/Provider';
-import { getFakeI18nInst } from 'tests/unit/helpers';
-
-
-class FakeChild extends React.Component {
-  render() {
-    return <p>The component</p>;
-  }
-}
-
-describe('Header', () => {
-  function renderHeader({ ...props }) {
-    const { store } = createStore();
-    store.dispatch(setClientApp('android'));
-    store.dispatch(setLang('en-GB'));
-    const fakeI18n = getFakeI18nInst();
-
-    return findRenderedComponentWithType(renderIntoDocument(
-      <Provider store={store}>
-        <I18nProvider i18n={fakeI18n}>
-          <HeaderBase i18n={fakeI18n} {...props} />
-        </I18nProvider>
-      </Provider>
-    ), HeaderBase);
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import { getFakeI18nInst, shallowUntilTarget } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  function renderHeader({
+    store = dispatchClientMetadata().store,
+    ...props
+  } = {}) {
+    return shallowUntilTarget(
+      <Header i18n={getFakeI18nInst()} store={store} {...props} />,
+      HeaderBase,
+    );
   }
 
   it('renders an <h1> when isHomepage is true', () => {
-    const root = renderHeader({
-      isHomePage: true,
-      children: FakeChild,
-      SearchFormComponent: FakeChild,
-    });
-    const headerTag = findDOMNode(root).querySelector('.Header-title-wrapper');
-
-    expect(headerTag.textContent).toEqual('Firefox Add-ons');
-    expect(headerTag.tagName).toEqual('H1');
+    const { store } = dispatchClientMetadata();
+    store.dispatch(setViewContext(VIEW_CONTEXT_HOME));
+    const root = renderHeader({ store });
+
+    expect(root.find('.Header-title-wrapper')).toHaveTagName('h1');
+    expect(root.find('.Header-title').type()).toEqual(Link);
+    expect(root.find('.Header-title').prop('children'))
+      .toContain('Firefox Add-ons');
   });
 
-  it('always renders a link in the header', () => {
-    const root = renderHeader({
-      children: FakeChild,
-      SearchFormComponent: FakeChild,
-    });
-    const titleLink = findDOMNode(root).querySelector('.Header-title');
-    const h1Tag = findDOMNode(root).querySelector('h1');
+  it('always renders a link in the header when not on homepage', () => {
+    const root = renderHeader();
 
-    expect(h1Tag).toBeFalsy();
-    expect(titleLink.textContent).toEqual('Firefox Add-ons');
-    expect(titleLink.tagName).toEqual('A');
+    // There shouldn't be an H1 in the header on pages that aren't the
+    // homepage; other routes will render their own, more relevant, H1 tags.
+    expect(root.find('h1')).toHaveLength(0);
+    expect(root.find('.Header-title').type()).toEqual(Link);
+    expect(root.find('.Header-title').prop('children'))
+      .toContain('Firefox Add-ons');
   });
 
   describe('mapStateToProps', () => {
     it('gets isHomePage from store', () => {
-      const { store } = createStore();
+      const { store } = dispatchClientMetadata();
```

Comment:
I think you can just delete this test entirely because you're already dispatching the same actions for the *renders an h1 when isHomepage is true* test.